### PR TITLE
build: Add static type checking (mypy) to CI

### PR DESCRIPTION
**Pull request by johevemi**
## References

* #565

### CODE REVIEW

File: .github/workflows/ci-databricks.yml [hunk 1]
```diff
@@ -34,6 +34,17 @@ jobs:
       IGNORE_ERRORS_AND_WARNING_FLAKE8: 'E501,F401,E402,E203,W503'
       TEST_REPORT_PATH: ./source/databricks/tests/htmlcov/
 
+  mypy_check:
+    runs-on: ubuntu-latest
+    name: Static type checker
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
```
Comment (djorgensendk) [inline .github/workflows/ci-databricks.yml:42]: v4 of this action is available

File: .github/workflows/ci-databricks.yml [hunk 2]
```diff
@@ -34,6 +34,17 @@ jobs:
       IGNORE_ERRORS_AND_WARNING_FLAKE8: 'E501,F401,E402,E203,W503'
       TEST_REPORT_PATH: ./source/databricks/tests/htmlcov/
 
+  mypy_check:
+    runs-on: ubuntu-latest
+    name: Static type checker
+    steps:
+    - uses: actions/checkout@v2
```
Comment (djorgensendk) [inline .github/workflows/ci-databricks.yml:41]: please use v3 of this action

File: .github/workflows/ci-databricks.yml [hunk 3]
```diff
@@ -34,6 +34,17 @@ jobs:
...
+    - uses: actions/setup-python@v2
+      with:
+        python-version: 3.x
+    - run: pip install --upgrade pip
```
Comment (djorgensendk) [inline .github/workflows/ci-databricks.yml:45]: can be merged into one action instead with something like:
- name: <meaningful name>
   shell: bash
   run: |
            <commands>

File: .github/workflows/ci-databricks.yml [hunk 4]
```diff
@@ -34,6 +34,17 @@ jobs:
       IGNORE_ERRORS_AND_WARNING_FLAKE8: 'E501,F401,E402,E203,W503'
       TEST_REPORT_PATH: ./source/databricks/tests/htmlcov/
 
+  mypy_check:
```
Comment (djorgensendk) [inline .github/workflows/ci-databricks.yml:37]: should the pipeline be able to succeed if the job fails? if not, make allow_merge depend on it by adding it to the needs-array

**Review by djorgensendk — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-databricks.yml (hunk 2 above)
Comment (johevemi) [reply .github/workflows/ci-databricks.yml]: updated to v3

File: .github/workflows/ci-databricks.yml (hunk 4 above)
Comment (johevemi) [reply .github/workflows/ci-databricks.yml:37]: added to needs-array

File: .github/workflows/ci-databricks.yml (hunk 3 above)
Comment (johevemi) [reply .github/workflows/ci-databricks.yml]: I have changed the code to match your suggestion, is the new code what you had in mind?

**Review by djorgensendk — APPROVED** [review]

**Pull request merged**